### PR TITLE
bpo-46827: pass sock.type to getaddrinfo in sock_connect

### PR DESCRIPTION
**Pull request by graingert**
https://bugs.python.org/issue46827

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2022-02-22-15-08-30.bpo-46827.hvj38S.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+support UDP sockets in  SelectorEventLoop.sock_connect
```
Comment (asvetlov) [inline Misc/NEWS.d/next/Library/2022-02-22-15-08-30.bpo-46827.hvj38S.rst]:
```suggestion
Support UDP sockets in  :meth:`asyncio.AbstractEventLopp.sock_connect` for selector-based event loops.  Patch by Thomas Grainger.
```

**Review by asvetlov — APPROVED** [review]
LGTM.
Would you consider changing NEWs record as suggested?

**Comment by graingert** [comment]
@asvetlov I think this needs backporting to 3.9 and 3.10?

File: Misc/NEWS.d/next/Library/2022-02-22-15-08-30.bpo-46827.hvj38S.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Support UDP sockets in  :meth:`asyncio.AbstractEventLopp.sock_connect` for selector-based event loops.  Patch by Thomas Grainger.
```
Comment (graingert) [inline Misc/NEWS.d/next/Library/2022-02-22-15-08-30.bpo-46827.hvj38S.rst]: I think the loop methods are in `asyncio.loop` in sphinx: https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.sock_connect

```suggestion
Support UDP sockets in  :meth:`asyncio.loop.sock_connect` for selector-based event loops.  Patch by Thomas Grainger.
```

**Comment by asvetlov** [comment]
The PR doesn't change the behavior but makes unsupported things work.
Thus I think the backporting is safe.

**Pull request merged**